presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/credential/ExtraCredentialProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc.credential;
+
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.JdbcIdentity;
+
+import javax.inject.Inject;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ExtraCredentialProvider
+        implements CredentialProvider
+{
+    private final Optional<String> userCredentialName;
+    private final Optional<String> passwordCredentialName;
+    private final CredentialProvider delegate;
+
+    @Inject
+    public ExtraCredentialProvider(BaseJdbcConfig baseJdbcConfig, @ForExtraCredentialProvider CredentialProvider delegate)
+    {
+        this(Optional.ofNullable(baseJdbcConfig.getUserCredentialName()), Optional.ofNullable(baseJdbcConfig.getPasswordCredentialName()), delegate);
```

Comment:
Can you please refactor `baseJdbcConfig.getPasswordCredentialName()` and `baseJdbcConfig.getUserCredentialName()` to return `Optional<String>`?